schema/appmodel/wiec.schema.xml

```diff
@@ -0,0 +1,185 @@
+  <attribute name="pulse_duration" description="WIB pulser pulse duration in system clock units (16 ns). 27 bits [0-134217727]" type="u32" init-value="255" is-not-null="yes"/>
+ </class>
+
+ <class name="WIBSettings" description="WIB system settings (argument to settings)">
+  <attribute name="cold" description="True if the front end electronics are COLD (77k)" type="bool" init-value="false" is-not-null="yes"/>
+  <attribute name="pulser" description="True if the calibration pulser should be enabled" type="bool" init-value="false" is-not-null="yes"/>
+  <attribute name="detector_type" description="Detector type selector: WIB default (0), upper APA (1), lower APA (2), CRP (3)" type="u8" range="0,1,2,3" init-value="0" is-not-null="yes"/>
+  <attribute name="adc_test_pattern" description="True if the COLDADC test pattern should be enabled" type="bool" init-value="false" is-not-null="yes"/>
+  <relationship name="femb0" description="Settings for FEMB in slot 0" class-type="FEMBSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="femb1" description="Settings for FEMB in slot 1" class-type="FEMBSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="femb2" description="Settings for FEMB in slot 2" class-type="FEMBSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="femb3" description="Settings for FEMB in slot 3" class-type="FEMBSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="coldadc_settings" description="Custom register settings for ColdADC" class-type="ColdADCSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="wib_pulser" description="Settings for WIB pulser" class-type="WIBPulserSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+ </class>
+
+ <class name="WIBModuleConf">
+  <attribute name="communication_type" type="string" init-value="tcp" is-not-null="yes"/>
+  <attribute name="communication_port" type="u32" init-value="1234" is-not-null="yes"/>
+  <relationship name="settings" class-type="WIBSettings" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+ </class>
+
+ <class name="WIECApplication" description="Application describing the controller of a detector unit using WIBs inside a WIEC.">
+  <superclass name="SmartDaqApplication"/>
+  <superclass name="ResourceSetAND"/>
+  <relationship name="wib_module_conf" class-type="WIBModuleConf" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+  <relationship name="hermes_module_conf" class-type="HermesModuleConf" low-cc="one" high-cc="one" is-composite="no" is-exclusive="no" is-dependent="no"/>
+ </class>
```

Comment:
This class should define a `generate_modules()` method. Without it,  src/WIECApplication.cpp will not compile.

Reply:
added the algo, compiled locally and didn't see any issue.